spring-cloud-stream/src/main/java/org/springframework/cloud/stream/micrometer/MessageChannelStatsdMetricsAutoConfiguration.java

```diff
@@ -16,65 +16,75 @@
 import org.springframework.boot.actuate.autoconfigure.metrics.export.simple.SimpleMetricsExportAutoConfiguration;
+import org.springframework.boot.actuate.autoconfigure.metrics.export.statsd.StatsdMetricsExportAutoConfiguration;
 import org.springframework.boot.autoconfigure.AutoConfigureAfter;
 import org.springframework.boot.autoconfigure.AutoConfigureBefore;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.cloud.stream.binder.Binder;
 import org.springframework.cloud.stream.binding.BindableProxyFactory;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.support.GenericMessage;
 
 /**
+ * Emits metrics in the Datadog StatsD line format over a message channel.
  *
  * @author Oleg Zhurakousky
+ * @author Jon Schneider
  *
  * @since 2.0
- *
  */
 @Configuration
-@AutoConfigureBefore(SimpleMetricsExportAutoConfiguration.class)
-@AutoConfigureAfter(MetricsAutoConfiguration.class)
+@AutoConfigureBefore({ CompositeMeterRegistryAutoConfiguration.class, SimpleMetricsExportAutoConfiguration.class })
+@AutoConfigureAfter({ MetricsAutoConfiguration.class, StatsdMetricsExportAutoConfiguration.class })
 @ConditionalOnClass({ Binder.class, MetricsAutoConfiguration.class })
 @ConditionalOnProperty("spring.cloud.stream.bindings." + MetersPublisherBinding.APPLICATION_METRICS + ".destination")
 @EnableConfigurationProperties(ApplicationMetricsProperties.class)
-public class DestinationPublishingMetricsAutoConfiguration {
+public class MessageChannelStatsdMetricsAutoConfiguration {
 
 	@Bean
-	@ConditionalOnMissingBean
-	public MetricsPublisherConfig metricsPublisherConfig(ApplicationMetricsProperties metersPublisherProperties) {
-		return new MetricsPublisherConfig(metersPublisherProperties);
-	}
+	public StatsdMeterRegistry statsdRegistry(ApplicationMetricsProperties applicationMetricsProperties,
+			MetersPublisherBinding publisherBinding, Clock clock) {
 
-	@Bean
-	@ConditionalOnMissingBean
-	public DefaultDestinationPublishingMeterRegistry defaultDestinationPublishingMeterRegistry(
-			ApplicationMetricsProperties applicationMetricsProperties,
-			MetersPublisherBinding publisherBinding,
-			MetricsPublisherConfig metricsPublisherConfig, Clock clock) {
-		return new DefaultDestinationPublishingMeterRegistry(applicationMetricsProperties, publisherBinding, metricsPublisherConfig, clock);
+		StatsdMeterRegistry registry = StatsdMeterRegistry.builder(StatsdConfig.DEFAULT)
+				.clock(clock)
+				.lineSink(line -> publisherBinding.applicationMetrics().send(new GenericMessage<String>(line)))
```

Comment:
This is what sinks StatsD lines on the output channel.

Reply:
John, i'll probably make consumer a bean with Conditional so users can override it